tests/unit/test_entities_souffler.py

```diff
@@ -0,0 +1,117 @@
+"""Unit test module for playlist_souffle.entities.souffler."""
```

Comment:
I'd put in the unit tests in the same directory as the files they're testing: entities/souffler.py and entities.test.py or something like that. Why? You see them together, you update them, easy to open at the same time, in PR's they are next to each other, you notice when they are missing.

(but any system tests that require services (including your own) to be up and running, I'd put in a separate integration tests (or system tests) directory (like you do)

Reply:
I'm going to work on testing in a future PR so will leave off until then.

But there are a few reasons to keep test structure as it is:
1. Its the most common / first recommended testing dir structure by pytest.
2. In pytest you can have directory specific conftest.py files for fixture setup shared between tests. Works well with the tests/ dir structure i think.
3. I'd have to add some logic somewhere to remove those files from the playlist_souffle package.
4. I generally like to keep my dir structure to represent the package structure.


Do agree that your suggestion has benefits. Lets talk more in the testing PR

Reply:
4) i mean, if my python package is

package_a
- package_a.module_x
- package_b.module_y

I like the dir structure to be just
package_a/
- __init__.py
- module_x.py
- module_y.py


2) i'm pretty sure with python conftest.py files, the configuration files need to be in the same directory or a parent directory of the test files themselves.

3) 
a - when i deploy the package to AWS, i just want to deploy the source code, not the test files. So would need to omit them from the directory tree rather than just not include the tests dir.
b- if we apply this to all of my python projects, let's say i'm building a generic set of aws python tools and want to deploy the package to pip. I'd need logic in my setup.py to omit all of those files from the distribution package. Again here, i think theres value in having the src/ code be the exact code that a user can expect in a package.

Checked requests and reitz uses the tests pytest directory structure. I'm pretty sure that is the standard and fits into the "in python there is one way to things" but i'd be curious if you could find a major python repo that has unit tests beside their source files.

I do think the mapping from source to test in this layout is not very clear